src/Xamarin.Android.Build.Tasks/Utilities/NuGetLogger.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using NuGet.Common;
+using Microsoft.Build.Utilities;
+
+namespace Xamarin.Android.Tasks {
+
+	class NuGetLogger : ILogger {
+		TaskLoggingHelper log;
+
+		public NuGetLogger (TaskLoggingHelper log)
+		{
+			this.log = log;
+		}
+
+		public void Log (LogLevel level, string data)
+		{
+			log.LogDebugMessage (data);
+		}
+
+		public void Log (ILogMessage message)
+		{
+			log.LogDebugMessage (message.Message);
+		}
+
+		public System.Threading.Tasks.Task LogAsync (LogLevel level, string data)
+		{
+			return System.Threading.Tasks.Task.Run (() => Log (level, data));
```

Comment:
Is this actually safe? See also what `AsyncTask` has to go through for async logging...

Reply:
ok probably not...not sure we want `ResolveAssemblies` to derive from `AsyncTask` though.